src/gameboy/mod.rs

```diff
@@ -96,7 +96,12 @@ impl Gameboy {
         'running: loop {
             self.handle_sdl2_events()?;
             self.handle_interrupts();
-            self.cpu.step();
+
+            if !self.cpu.halted() {
+                self.cpu.step();
+            } else {
+                sleep(Duration::from_millis(10));
```

Comment:
Leave a comment saying this may dispensary.

Reply:
![image](https://user-images.githubusercontent.com/13443805/174505173-bb1c74a5-e04e-49cd-8daa-b7fa09809d30.png)

Reply:
done